mongo/options/autoencryptionoptions.go

```diff
@@ -91,6 +96,38 @@ func (a *AutoEncryptionOptions) SetExtraOptions(extraOpts map[string]interface{}
 	return a
 }
 
+// SetTLSOptions specifies tls.Config instances for each KMS provider to use to configure TLS on all connections created
+// to the cluster. The input map should contain a mapping from each KMS provider to a document containing the necessary 
+// options, as follows:
+//
+// {
+//		"kmip": {
+//			"tlsCertificateKeyFile": "foo.pem",
+// 			"tlsCAFile": "fooCA.pem"
+//		}
+// }
+//
+// Currently, the following TLS options are supported:
+//
+// 1. "tlsCertificateKeyFile" (or "sslClientCertificateKeyFile"): The "tlsCertificateKeyFile" option specifies a path to 
+// the client certificate and private key, which must be concatenated into one file. 
+//
+// 2. "tlsCertificateKeyFilePassword" (or "sslClientCertificateKeyPassword"): Specify the password to decrypt the client
+// private key file (e.g. "tlsCertificateKeyFilePassword=password").
+//
+// 3. "tlsCaFile" (or "sslCertificateAuthorityFile"): Specify the path to a single or bundle of certificate authorities
+// to be considered trusted when making a TLS connection (e.g. "tlsCaFile=/path/to/caFile").
+//
+// This should only be used to set custom TLS options. By default, the connection will use an empty tls.Config{}.
+func (a *AutoEncryptionOptions) SetTLSOptions(tlsOpts map[string]map[string]interface{}) (*AutoEncryptionOptions, error) {
+	a.TLSConfig = make(map[string]*tls.Config)
+	_, err := applyTLSOptions(tlsOpts, a.TLSConfig)
```

Comment:
Hmm should you do `a.TLSConfig, err := applyTLSOptions(tlsOpts, a.TLSConfig)` here and in `clientencryptionoptions.go`? I'm not sure the `tlsConfigs` param of `applyTLSOptions` is getting passed by reference.

Reply:
Also not sure if it's getting passed by reference, although the tests are passing. Assigning the output of `applyTLSOptions()` to the value of TLSConfig now just in case